### DISCUSSION TITLE
KubeCPUOvercommit  Alert message field not populated in metalk8s ui

### DISCUSSION THREAD

**Discussion by n1603** [scality/metalk8s#2993] [General]
I have installed metalk8s and all seems to work fine, except for Alert UI. I don;t see alert message for some of the alert like KubeCPUOvercommit. 

I verified some data, like ./charts/kube-prometheus-stack/templates/prometheus/rules/kubernetes-resources.yaml have message field under annotations. but ./charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-resources.yaml do not have it.

Not sure which one is picked. Could you please suggest any configuration I could have missed.

The prometheus config has below file pointed and it does not have message filed. 
/etc/prometheus/rules/prometheus-prometheus-operator-prometheus-rulefiles-0/metalk8s-monitoring-prometheus-operator-kubernetes-resources.yaml

- alert: KubeCPUOvercommit
    annotations:
      description: Cluster has overcommitted CPU resource requests for Pods and cannot tolerate node failure.
      runbook_url: https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/runbook.md#alert-name-kubecpuovercommit
      summary: Cluster has overcommitted CPU resource requests.

**Comment by gdemonet** [comment]
Hello @n1603, thank you for reporting this. It looks like an actual bug from our UI code, so if you don't mind, I'll open an issue to track this bug.

There is unfortunately nothing you missed when configuring your cluster: this alert doesn't have a `message` field when fired, and our UI (currently, in 2.7.0-dev and lower) only displays such field (whereas, to display useful information, it would need to show the `description` field, in this case).

I will keep you posted once a fix lands!

**Reply by gdemonet** [reply]
(See #2994)

**Reply by n1603** [reply]
Hi @gdemonet 

Thanks for looking into this. 

On below :
>>  "this alert doesn't have a message field when fired"

I verified this part, thats correct. Thanks

One additional information here which I missed before is I am running metalk8s-2.6 version and moved to helm3 and upgraded to prometheus  12.2.3 (also tried 12.8). 

Can you please give some idea of the flow on UI side for this issue, if it is straight forward, I will try to  fix and test.

**Reply by n1603** [reply]
I tried to fix this issue like this : [link](https://github.com/n1603/metalk8s/blob/development/2.7/ui/src/containers/ClusterMonitoring.js#L173)

**Comment by ChengYanJin** [comment]
Hello Ganesh(@n1603),

Thanks for considering contributing to MetalK8s!
 
I will explain to you the steps so we can work together to eventually merge your fix.
* Please refer to our [commit message best practice](https://metal-k8s.readthedocs.io/en/latest/developer/development/commit.html?highlight=commit) to format yours.


* Once your branch is clean, you may want to create a **pull request**.
* We will do the **review** and may request some changes. 
* Once we’re all good, we'll clone the branch and submit it to the main repo, to get CI validation, and then merge it.

Please let us know if there is anything unclear.
Thank you

**Reply by n1603** [reply]
sure @ChengYanJin, I will take a look at the process.
